benchmarking/lance/test_query.py

```diff
@@ -0,0 +1,149 @@
+            return df.select("image").collect()
+
+    result = benchmark.pedantic(operation, rounds=5, warmup_rounds=1)
+
+    if agg_name == "count":
+        assert result == 1000
+    elif agg_name == "sum_struct":
+        result_data = result.to_pydict()
+        print(f"daft height: { result_data['height'][0]}")
+        assert result_data["height"][0] > 0
+    elif agg_name == "sum_int":
+        result_data = result.to_pydict()
+        print(f"daft size: { result_data['size'][0]}")
+        assert result_data["size"][0] > 0
+
+
+@pytest.mark.parametrize("agg_name, _", [case.values[:2] for case in TEST_CASES])
+def test_daft_sql_read_lance(benchmark, agg_name, _):
+    """Benchmark Daft aggregations on Lance data."""
+
+    def operation():
+        df = daft.read_lance(PATH)
+        if agg_name == "count":
+            return df.count_rows()
+        elif agg_name == "sum_struct":
+            df = daft.sql("SELECT sum(image.height) FROM df")
+            return df.collect()
+        elif agg_name == "sum_int":
+            df = daft.sql("SELECT sum(size) FROM df")
+            return df.collect()
+        else:
+            df = daft.sql("SELECT image FROM df")
+            return df.collect()
+
+    result = benchmark.pedantic(operation, rounds=5, warmup_rounds=1)
+
+    if agg_name == "count":
+        assert result == 1000
+    elif agg_name == "sum_struct":
+        result_data = result.to_pydict()
+        print(f"daft height: { result_data['height'][0]}")
+        assert result_data["height"][0] > 0
+    elif agg_name == "sum_int":
+        result_data = result.to_pydict()
+        print(f"daft size: { result_data['size'][0]}")
+        assert result_data["size"][0] > 0
+
+@pytest.mark.parametrize("agg_name, _", [case.values[:2] for case in TEST_CASES])
+def test_lance_native_read(benchmark, agg_name, _):
+    """Benchmark Lance native query execution."""
+
+    def read_fn():
+        df = lance.dataset(PATH)
+        if agg_name == "count":
+            return df.count_rows()
+        elif agg_name == "sum_struct":
+            query = df.sql("SELECT sum(image.height) FROM dataset").build()
+            return query.to_batch_records()
+        elif agg_name == "sum_int":
+            query = df.sql("SELECT sum(size) FROM dataset").build()
+            return query.to_batch_records()
+        else:
+            return df["image"]
+
+    result = benchmark(read_fn)
+
+    if agg_name == "count":
+        assert result == 1000
+    elif agg_name == "sum_struct":
+        height = result[0].to_pydict()["sum(dataset.image[height])"][0]
+        print(f"native height: {height }")
+        assert height > 0
+    elif agg_name == "sum_int":
+        size = result[0].to_pydict()["sum(dataset.size)"][0]
+        print(f"native size: {size} ")
+        assert size > 0
+    else:
+        assert len(result) == 1000
+
+
+@pytest.mark.parametrize("agg_name, agg_expr", TEST_CASES)
+def test_duckdb_read_lance(benchmark, agg_name, agg_expr):
```

Comment:
this duckdb benchmark seems a bit unfair as it's not using lance, but arrow, and the lance to arrow conversion happens outside of the benchmark.